daemon/loop.go

```diff
@@ -21,40 +21,74 @@ type LoopVars struct {
 	GitPollInterval      time.Duration
 	RegistryPollInterval time.Duration
 	syncSoon             chan struct{}
-	initSyncSoon         sync.Once
+	pollImagesSoon       chan struct{}
+	initOnce             sync.Once
+}
+
+func (loop *LoopVars) ensureInit() {
+	loop.initOnce.Do(func() {
+		loop.syncSoon = make(chan struct{}, 1)
+		loop.pollImagesSoon = make(chan struct{}, 1)
+	})
 }
 
 // Loop for potentially long-running stuff. This includes running
 // jobs, and looking for new commits.
 
 func (d *Daemon) Loop(stop chan struct{}, wg *sync.WaitGroup, logger log.Logger) {
 	defer wg.Done()
-	pollGit := time.NewTimer(d.GitPollInterval)
+	gitPollTimer := time.NewTimer(d.GitPollInterval)
+	// This can be called any time to push the next git poll out by
+	// d.GitPollInterval. It must _at least_ be called when the timer
+	// fires, so that the polling continues.
 	resetGitPoll := func() {
```

Comment:
Yeahp, style difference :) Let me explain my thinking a bit, for posterity.

It is mainly a matter of what's convenient.

`doSync` is long*, though reasonably self-contained, and aside from its receiver, only needs a logger. It's convenient to be able to use it as a value.

`pullThen` is much shorter, and used like a local macro (i.e., not relevant anywhere else). It's convenient to close over the logger and daemon rather than supplying them as arguments. But it's not _toooo_ much different as a method, and if it were useful to call from elsewhere, it could easily become one.

*we should break it down into its parts! later ...